docs/developers/rest.md

```diff
@@ -116,35 +116,119 @@ GET /my-resource/?age=le=20
+This function allows you to specify which properties should be included in the responses. This takes several forms:
 
 * `?select(property)`: This will return the values of the specified property directly in the response (will not be put in an object).
 * `?select(property1,property2)`: This returns the records as objects, but limited to the specified properties.
 * `?select([property1,property2,...])`: This returns the records as arrays of the property values in the specified properties.
 * `?select(property1,)`: This can be used to specify that objects should be returned with the single specified property.
+* `?select(property{subProperty1,subProperty2{subSubProperty,..}},...)`: This can be used to specify which sub-properties should be included in nested objects and joined/references records.
 
 To get a list of product names with a category of software:
 
 ```http
-GET /product/?category=software&select(name)
+GET /Product/?category=software&select(name)
 ```
 
 ### `limit(start,end)` or `limit(end)`
 
-Specifies a limit on the number of records returned, optionally providing a starting offset.
+This function specifies a limit on the number of records returned, optionally providing a starting offset.
 
 For example, to find the first twenty records with a `rating` greater than 3, `inStock` equal to true, only returning the `rating` and `name` properties, you could use:
 
 ```http
-GET /product?rating=gt=3&inStock=true&select(rating,name)&limit(20)
+GET /Product/?rating=gt=3&inStock=true&select(rating,name)&limit(20)
+```
+
+### `sort(property)`, `sort(+property,-property,...)`
+
+This function allows you to indicate the sort order for the returned results. The argument for `sort()` is one or more properties that should be used to sort. If the property is prefixed with '+' or no prefix, the sort will be performed in ascending order by the indicated attribute/property. If the property is prefixed with '-', it will be sorted in descending order. If the multiple properties are specified, the sort will be performed on the first property, and for records with the same value for that property, the next property will be used to break the tie and sort results. This tie breaking will continue through any provided properties.
+
+For example, to sort by product name (in ascending order):
+```http
+GET /Product?rating=gt=3&sort(+name)
+```
+To sort by rating in ascending order, then by price in descending order for products with the same rating:
+```http
+GET /Product?sort(+rating,-price)
+```
+
+## Nested Attributes and Joins
+In addition to querying on top-level attributes, you can also query on nested attributes. Most importantly, this provides "join" functionality, allowing related tables to be queried and joined in the results. Nested properties are specified by using dot syntax. In order to effectively leverage join functionality, you need to define a relationship in your schema:
```

Comment:
I read nested attributes as:
```
{
  "name": "Harper",
  "traits": {
    "goofy": true,
    "smart": true,
    "crazy" true
  }
}
```
Meaning it's nested JSON. So for this, I might call it joined attributes or something like that? I'm not exactly sure, just giving you my immediate impression. It still makes sense once I read it all the way through.

Reply:
Maybe "chained attributes"?